woocommerce/woocommerce-ios
# [ABTesting] Cache variations of logged out context experiments

Closes: #8782 


- [ ] ⚠️ Kindly review https://github.com/Automattic/Automattic-Tracks-iOS/pull/247

## Description

Raising this PR based on the solution we reached after discussions in this PR - https://github.com/woocommerce/woocommerce-ios/pull/8788

We are caching the variation values of logged-out experiments to try to address the high crossovers that we see in Explat.


As this PR targets the release branch, I have cherry-picked commits from this PR #8769. This is done to import `ABTestVariationProvider`

## Testing instructions
- Install and launch the app. Log out if needed.
- Launch and skip onboarding.
- If you don't see the "Continue With WordPress.com" button to log in you have been assigned `treatment` variation for the `woocommerceios_login_rest_api_project_202301_v2` experiment and you can test this PR (If not, you need to use a sandbox to force a variation. You can follow the instructions from [this PR](https://github.com/woocommerce/woocommerce-ios/pull/8744) to assign a variation to your device.)
- Login into the app using site creds.
- Go to settings and log out.
- You will be navigated to the login screen and ensure that you still don't see the "Continue With WordPress.com" button. i.e. You are still assigned the `treatment` variation.

## Screenshots
NA

---
